.build/tasks/Merge-CodeCoverageFiles.pester.build.ps1

```diff
@@ -0,0 +1,139 @@
+Param (
+    # Project path
+    [Parameter()]
+    [string]
+    $ProjectPath = (property ProjectPath $BuildRoot),
+
+    [Parameter()]
+    # Base directory of all output (default to 'output')
+    [string]
+    $OutputDirectory = (property OutputDirectory (Join-Path $BuildRoot 'output')),
+
+    [Parameter()]
+    [string]
+    $ProjectName = (property ProjectName $(
+            (Get-ChildItem $BuildRoot\*\*.psd1 -Exclude 'build.psd1', 'analyzersettings.psd1' | Where-Object {
+                    ($_.Directory.Name -match 'source|src' -or $_.Directory.Name -eq $_.BaseName) -and
+                    $(try
+                        {
+                            Test-ModuleManifest $_.FullName -ErrorAction Stop
+                        }
+                        catch
+                        {
+                            Write-Warning $_
+                            $false
+                        }) }
+            ).BaseName
+        )
+    ),
+
+    [Parameter()]
+    [string]
+    $ModuleVersion = (property ModuleVersion $(
+            try
+            {
+                (gitversion | ConvertFrom-Json -ErrorAction Stop).InformationalVersion
+            }
+            catch
+            {
+                Write-Verbose "Error attempting to use GitVersion $($_)"
+                ''
+            }
+        )),
+
+    # Build Configuration object
+    [Parameter()]
+    $BuildInfo = (property BuildInfo @{ })
+)
+
+. $PSScriptRoot/Common.Functions.ps1
+
+# Synopsis: Making sure the Module meets some quality standard (help, tests).
+task Merge_CodeCoverage_Files {
+    if (!(Split-Path -isAbsolute $OutputDirectory))
+    {
+        $OutputDirectory = Join-Path -Path $ProjectPath -ChildPath $OutputDirectory
+        Write-Build Yellow "Absolute path to Output Directory is $OutputDirectory"
+    }
+
+    $CodeCovOutputFile = "CodeCov_Merged.xml"
+    if ($BuildInfo.ContainsKey("Pester") -eq $true -and
+        $BuildInfo.Pester.ContainsKey("CodeCoverageMergedOutputFile") -eq $true)
+    {
+        $CodeCovOutputFile = $BuildInfo.Pester.CodeCoverageMergedOutputFile
+    }
+
+    $targetFile = Join-Path -Path $OutputDirectory -ChildPath $CodeCovOutputFile
+
+    if (Test-Path -Path $targetFile)
+    {
+        Write-Build Yellow "File $targetFile found, deleting file"
+        Remove-Item -Path $targetFile -Force
+    }
+
+    Write-Build White "Processing folder: $OutputDirectory"
+
+    $codecovFiles = Get-ChildItem -Path $OutputDirectory -Include 'codecov*.xml' -Recurse
+
+    if ($codecovFiles.Count -gt 1)
+    {
+        $firstFile = $codecovFiles | Select-Object -First 1
+        $otherFiles = $codecovFiles | Select-Object -Skip 1
+
+        Write-Build DarkGray "Started merging $($codecovFiles.Count) code coverage files!"
+        [xml]$targetDocument = Get-Content $firstFile.FullName
```

Comment:
You should do a Get-Content -Raw or create the XML object and using the Load() method. Much faster than doing Get-Content.

Reply:
Will update to add the Raw parameter. I have tried the XmlDocument.Load method, but since the file contains a DTD which we do not have, that method throws an error.